yt/yt/server/job_proxy/stderr_writer.cpp

```diff
@@ -142,12 +144,54 @@ void TStderrWriter::DoWrite(const void* buf_, size_t len)
     Tail_->Write(buf, len);
 }
 
-TString TStderrWriter::GetCurrentData() const
+NApi::TGetJobStderrResponse TStderrWriter::GetCurrentData(const NApi::TGetJobStderrOptions& options) const
 {
     TStringStream stringStream;
     stringStream.Reserve(GetCurrentSize());
-    SaveCurrentDataTo(&stringStream);
-    return stringStream.Str();
+    auto limit = options.Limit.value_or(0);
+    auto offset = options.Offset.value_or(0);
+    SaveCurrentDataTo(&stringStream, offset || limit);
+    auto data = stringStream.Str();
+    auto endOffset = TotalSize_;
+    const i64 currentFirstAbsolutePos = TotalSize_ > static_cast<i64>(data.size()) ? TotalSize_ - data.size() : 0;
```

Comment:
Am I understand correctly that we ignore first block of stderr in this logic?

Reply:
If we have only Head_ - we use Head_
If we have both - we use Tail_
This place can be improved for some cases, for example when requested range fits into head,
but code will be complicated with strange effects when requested range contains ..skipped.. area